### ISSUE TITLE
Document computed columns in select parameter

### ISSUE THREAD

**Issue by allentc** [begriffs/postgrest#443]
Postgrest supports `computed columns' for both ordering and filtering results.  It seems natural to also support computed columns among the elements of the select parameter.

**Comment by begriffs** [comment]
If you have a table `foo` with computed column `bar`, what do you get back when you make a get request to `/foo?select=*,bar` ?

**Comment by allentc** [comment]
I agree.  That is my intended use case.

I'd like to share two of of several cases I've encountered; postal address and full name.  Computed columns exist in the schema to synthesize a value for each of these by composing other columns (e.g. [simplified] full_name = last_name ++ first_name ++ etc., full_addr = ... ++ city ++ state ++ zip.)  The client application may then order and filter these computed values in a natural manner.  But the composition must be reimplemented on the client for display in a tabular view.  Obviously I'd like to do away with the client side reimplementation, which is both a maintenance burden and bug prone in the sense that it will tend diverge from the database implementation in subtle ways.  

One can argue that the same thing can be achieved now with a database view.  Yet the ordering and filtering affordances exist despite also being achievable via views.  Why not select as well?  Thus my issue.

Incidentally I've been studying Haskell since February with an eye to contributing this feature to PostgREST.  I can't commit to a deadline, but unless some urgency has emerged I'd still like to contribute this myself.

**Comment by begriffs** [comment]
@allentc really appreciate you learning Haskell to help out!

Now about adding a full_name column to your table, I think you can do it without making a whole view. If you define a function with the right name and argument type then postgresql will let you use it as a computed column. For instance:

```sql
CREATE FUNCTION full_name(users) RETURNS varchar
    LANGUAGE sql STABLE
    AS $_$ SELECT $1.first_name || ' ' || $1.last_name $_$;
```

Once this function exists the client can use it as if it were a column like this

```
GET /users?select=full_name,*
```

In fact the fake column is eligible for sorting too:

```
GET /users?select=full_name,*&order=full_name
```

Perhaps I misunderstood your use case though, let me know.

**Comment by ruslantalpa** [comment]
This works as @begriffs explained and can be closed

**Comment by begriffs** [comment]
@allentc can you confirm whether this solved the problem for you?

**Comment by allentc** [comment]
Yes, this information addresses my use case.  However, I've had a look through the current PostgREST documentation at both the github wiki pages (postgrest/wiki/Advanced-Endpoints and postgrest/wiki/Routing) and the postgrest.com/api/reading page, and I did not find a mention of the above or any examples of these forms.

The use of {*} is documented for embedding foreign entities, but no examples appear of the use shown above; i.e. the select=*,... case.  The use of a computed column in both filter expressions and for ordering is mentioned, but not in a select= expression.

It may seem self evident that these expressions are supported, but I know that at one point I experimented with formulating a read request containing a computed column and my attempt failed with some error (certainly because I somehow guessed wrong at the syntax,) and this convinced me that the capability wasn't supported.  So I definitely think it is worthwhile to call out these features in the documentation.
